src/Runner/Runner.php

```diff
@@ -297,4 +461,31 @@ private function dispatchEvent(string $name, Event $event): void
 
         $this->eventDispatcher->dispatch($event, $name);
     }
+
+    private function getFileIterator(): LintingResultAwareFileIteratorInterface
+    {
+        if (null === $this->fileIterator) {
+            throw new \RuntimeException('File iterator is not configured. Pass paths during Runner initialisation or set them after with `setFileIterator()`.');
+        }
+
+        $fileIterator = new \ArrayIterator(
+            $this->fileIterator instanceof \IteratorAggregate
+                ? $this->fileIterator->getIterator()
+                : $this->fileIterator
+        );
+
+        // In order to determine the amount of required workers, we need to know how many files we need to analyse
+        $this->fileCount = \count(iterator_to_array($fileIterator));
```

Comment:
imagine that finder is having heavy filtering like "if file contains regex".

we already do `iterator_to_array` on fileIterator, ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Console/Command/FixCommand.php#L269

Reply:
I know, this probably should be improved. The problem here is that `FixCommand` requires simple finder/iterator and needs to determine the amount of files for progress output, while `Runner` converts that iterator to linting iterator and needs the file count to determine amount of workers to spawn. I made [simple change](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777/commits/d72fb361fccd8a38b5fb0874795f7075b39b9217), it should improve performance since files count won't be determine for workers' chunks (not needed), only for the main process - let me know what you think.